### ISSUE TITLE
YouTube API v3

### ISSUE THREAD

**Issue by MrSaints** [ClaudeBot/hubot-youtube-info#5]
**New API endpoint as the old one is deprecated:**

`GET https://www.googleapis.com/youtube/v3/videos?part=contentDetails%2Cstatistics%2Csnippet&id={VIDEO_ID_HERE}&key={YOUR_API_KEY}`

It's part of the `videos.list` [resource](https://developers.google.com/youtube/v3/docs/videos/list) so it returns a _list of items_. Thus, to access the number of views for example: 

``` js
data.items[0].statistics.viewCount
```

Assuming `data` is the fetched / parsed JSON from the API.

Sample return from API call:

``` json
{
 "kind": "youtube#videoListResponse",
 "etag": "\"tbWC5XrSXxe1WOAx6MK9z4hHSU8/FVju79dVVBMGXdQVZBiE9geOys8\"",
 "pageInfo": {
  "totalResults": 1,
  "resultsPerPage": 1
 },
 "items": [
  {
   "kind": "youtube#video",
   "etag": "\"tbWC5XrSXxe1WOAx6MK9z4hHSU8/7-TwPYToOeOT84MVTtjy5cVvabc\"",
   "id": "tGx-u-Rmzws",
   "snippet": {
    "publishedAt": "2015-05-06T21:00:29.000Z",
    "channelId": "UC18NaGQLruOEw65eQE3bNbg",
    "title": "Dota 2 WTF Moments 115",
    "description": "Dota 2 fail/win compilation\nSubmit your clip / Manda tu video: \nhttp://dotawatafak.com/\n\nCheck out the WTF Tees:\nhttp://shrsl.com/?~8c3r\n\nFacebook\nhttps://www.facebook.com/DotaWatafak\n\nTwitter:\nhttps://twitter.com/Dota2WTF\n\n---------------\n\nLAST CLIP MUSIC\n\nTrack Name: OMFG - Hello\nVideo Link: https://www.youtube.com/watch?v=XRgiNUtJurk\nYoutube: https://www.youtube.com/channel/UCMkh-WZrlhqwh_m42ZQNe7g\nDownload: http://bit.ly/alexomfghello\nFacebook: https://www.facebook.com/alexomfg\nSoundCloud: https://soundcloud.com/alexomfg\nTwitter: https://twitter.com/AlexSavageOMFG",
    "thumbnails": {
     "default": {
      "url": "https://i.ytimg.com/vi/tGx-u-Rmzws/default.jpg",
      "width": 120,
      "height": 90
     },
     "medium": {
      "url": "https://i.ytimg.com/vi/tGx-u-Rmzws/mqdefault.jpg",
      "width": 320,
      "height": 180
     },
     "high": {
      "url": "https://i.ytimg.com/vi/tGx-u-Rmzws/hqdefault.jpg",
      "width": 480,
      "height": 360
     },
     "standard": {
      "url": "https://i.ytimg.com/vi/tGx-u-Rmzws/sddefault.jpg",
      "width": 640,
      "height": 480
     },
     "maxres": {
      "url": "https://i.ytimg.com/vi/tGx-u-Rmzws/maxresdefault.jpg",
      "width": 1280,
      "height": 720
     }
    },
    "channelTitle": "Dota Watafak",
    "categoryId": "20",
    "liveBroadcastContent": "none",
    "localized": {
     "title": "Dota 2 WTF Moments 115",
     "description": "Dota 2 fail/win compilation\nSubmit your clip / Manda tu video: \nhttp://dotawatafak.com/\n\nCheck out the WTF Tees:\nhttp://shrsl.com/?~8c3r\n\nFacebook\nhttps://www.facebook.com/DotaWatafak\n\nTwitter:\nhttps://twitter.com/Dota2WTF\n\n---------------\n\nLAST CLIP MUSIC\n\nTrack Name: OMFG - Hello\nVideo Link: https://www.youtube.com/watch?v=XRgiNUtJurk\nYoutube: https://www.youtube.com/channel/UCMkh-WZrlhqwh_m42ZQNe7g\nDownload: http://bit.ly/alexomfghello\nFacebook: https://www.facebook.com/alexomfg\nSoundCloud: https://soundcloud.com/alexomfg\nTwitter: https://twitter.com/AlexSavageOMFG"
    }
   },
   "contentDetails": {
    "duration": "PT8M11S",
    "dimension": "2d",
    "definition": "hd",
    "caption": "false",
    "licensedContent": true
   },
   "statistics": {
    "viewCount": "301",
    "likeCount": "5198",
    "dislikeCount": "36",
    "favoriteCount": "0",
    "commentCount": "364"
   }
  }
 ]
}
```

**Migration notes:**
- `contentDetails` contains the video duration (in [ISO6801](https://en.wikipedia.org/wiki/ISO_8601#Durations) format).
- `statistics` contains the views, likes / dislikes, favourites, comments.
- `snippet` contains the title, description, etc.

More on properties: https://developers.google.com/youtube/v3/docs/videos

**Comment by mbwk** [comment]
Thanks, I'll have a closer look at this over the weekend.

**Comment by mbwk** [comment]
Resolved as of 2f7e92f
